draft-yasskin-dispatch-bundled-exchanges.md

```diff
@@ -0,0 +1,675 @@
+matching response stream. Defined by Section 8 of {{!RFC7540}}.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED",
+"MAY", and "OPTIONAL" in this document are to be interpreted as
+described in BCP 14 {{!RFC2119}} {{!RFC8174}} when, and only when, they
+appear in all capitals, as shown here.
+
+## Mode of specification {#mode}
+
+This specification defines how conformant bundle parsers work. It does not
+constrain how encoders produce a bundle: although there are some guidelines in
+{{authoring-guidelines}}, encoders MAY produce any sequence of bytes that a
+conformant parser would parse into the intended semantics.
+
+This specification uses the conventions and terminology defined in the Infra
+Standard ({{INFRA}}).
+
+# Semantics {#semantics}
+
+A bundle is logically a set of HTTP exchanges, with a URL identifying the
+manifest(s) of the bundle itself. A client uses content negotiation to pick a
+particular response for that URL, often an App Manifest ({{appmanifest}}), which
+then identifies other significant exchanges within the bundle, like a start
+page.
+
+While the order of the exchanges is not semantically meaningful, it can
+significantly affect performance when the bundle is loaded from a network
+stream.
+
+A bundle is parsed from a stream of bytes, which is assumed to have the attributes and operations described in {{stream-operations}}:
+
+Bundle parsers support two operations, defined in {{semantics-load-metadata}}
+and {{semantics-load-response}} each of which can return an error instead of
+their normal result.
+
+## Stream attributes and operations {#stream-operations}
+
+* A sequence of **available bytes**. As the stream delivers bytes, these are
+  appended to the available bytes.
+* An **EOF** flag that's true if the available bytes include the entire stream.
+* A **current offset** within the available bytes.
+* A **wait for N bytes** operation, which blocks until the available bytes
+  include at least N bytes past the current offset. A wait operation fails if
+  the stream ends before enough bytes are received, either due to a network
+  error or because the stream has a finite length.
+* A **seek** operation to change the current offset, relative to either the
+  beginning of the available bytes or to the old current offset. A seek past the
+  end of the available bytes is an *error in this specification*.
+* A **read N bytes** operation, which returns the sequence of N bytes starting
+  at the current offset, and then seeks forward by N bytes. A read past the end
+  of the available bytes is also an error in this specification.
+
+## Load a bundle's metadata {#semantics-load-metadata}
+
+This takes the bundle's stream and returns a struct ({{INFRA}}) of metadata
+containing at least items named:
+
+requests
+
+: A map ({{INFRA}}) whose keys are the HTTP requests ({{FETCH}}) for the
```

Comment:
What Fetch defines as "requests" are not necessarily HTTP requests. They're more broad.

Reply:
Even those are more broad as they carry fields HTTP requests don't have. I guess you could say requests whose url list contains a single URL whose scheme is an HTTP(S) scheme (unless this is restricted to secure contexts?).

Reply:
I've removed the attempt to subset the kinds of Fetch requests this operation can return. If it would noticeably simplify the request-matching function I need to write, I might add the subsetting back later, but I don't really expect that.